tilelang/profiler/__init__.py

```
@@ -57,13 +58,41 @@ def with_default_adapter(self, adapter: BaseKernelAdapter) -> Profiler:
+        """
+        new_shape = []
+        for dim in param.shape:
+            if isinstance(dim, tir.Var):
+                var_name = dim.name
+                if var_name in constraints:
+                    new_shape.append(constraints[var_name])
+                else:
+                    raise ValueError(
+                        f"Dynamic symbolic variable '{var_name}' not found in constraints. "
+                        f"Available constraints: {list(constraints.keys())}"
+                    )
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)